src/instance.rs

```diff
@@ -657,20 +657,28 @@ impl<T> Py<T> {
     /// Calls a method on the object.
     ///
     /// This is equivalent to the Python expression `self.name(*args, **kwargs)`.
-    pub fn call_method(
+    ///
+    /// To avoid repeated temporary allocations of Python strings, the [`intern!`] macro can be used
+    /// to intern `name`.
+    pub fn call_method<N, A>(
         &self,
         py: Python<'_>,
-        name: &str,
-        args: impl IntoPy<Py<PyTuple>>,
+        name: N,
+        args: A,
         kwargs: Option<&PyDict>,
-    ) -> PyResult<PyObject> {
+    ) -> PyResult<PyObject>
+    where
+        N: IntoPy<Py<PyString>>,
+        A: IntoPy<Py<PyTuple>>,
+    {
+        let name: Py<PyString> = name.into_py(py);
+        let args: Py<PyTuple> = args.into_py(py);
+
+        let ptr = self.getattr(py, name)?.into_ptr();
+        let args = args.into_ptr();
+        let kwargs = kwargs.into_ptr();
+
         unsafe {
-            let args = args.into_py(py).into_ptr();
-            let kwargs = kwargs.into_ptr();
-            let ptr = ffi::PyObject_GetAttr(self.as_ptr(), name.to_object(py).as_ptr());
```

Comment:
Sorry this is a bit messy since resolving the merge conflict.

How do you feel about:

```rust
pyo3::ffi::Something(name.to_object(py).as_ptr());
```

versus 

```rust
let ptr =  name.to_object(py).into_ptr();
pyo3:;ffi::Something(ptr);
ffi::Py_DECREF(ptr);
```
(or something else?)

Personally I prefer the latter. IMO the former is error prone and gives a bad example.

Reply:
> Why do you feel the former is error prone? I think the second is more error prone because missing the `Py_DECREF` call is leaking memory just like missing `free()` in C.

I'm mostly worried about someone reading it and and writing it slightly differently:
```rust
let ptr = name.to_object(py).as_ptr();
pyo3::ffi::Something(ptr);
```
which produces a dangling pointer, as the python object is dropped at the end of the statement (I should add documentation about this).

> Perhaps a middle ground is to make it obvious we're using RAII through `Py`?
> 
> ```rust
> let name: PyObject = name.to_object(py);
> pyo3::ffi::Something(name.as_ptr());
> ```

Sure 👍

Reply:
> 'm mostly worried about someone reading it and and writing it slightly differently:

This is an excellent point which I'd totally missed. IMO the middle ground above is probably the only sensible way to arrange this then. If you like, I'll submit a PR to fix the whole codebase to use this pattern once this PR merged? That way you can just worry about the modified functions here without another merge conflict.